### ISSUE TITLE
[css-cascade-6] Handle sibling-proximity in @scope

### ISSUE THREAD

**Issue by tabatkins** [w3c/csswg-drafts#7751]
Branching off of #7233...

Currently, `>>` is a shorthand for writing an `@scope`. That is, anytime you see `.foo >> .bar {...}`, it's equivalent to `@scope .foo { .bar {...}}`.

Now that we've agreed to add `++` (or `~~`) to do sibling proximity, it would be nice to keep the consistency between the two syntaxes.

I suggest `@scope siblings ... {}`, where the `...` is the current scope prelude stuff. If you specify `siblings` the lower bound, and all contained selectors, are auto-scoped to be siblings of the upper bound element, and we apply sibling-distance proximity as the tiebreaker rather than ancestor-distance proximity.

Need to be careful with the definitions to make sure descendant-scopes can still be nested; I guess descendant-scopes would scope the upper bound as a sibling but switch back to normal scoping for their lower bound. I believe this'll reproduce the behavior of `.foo >> .bar ++ .baz >> .qux`.

**Comment by ByteEater-pl** [comment]
Shouldn't that be `@scope (.foo) { .bar {...}}` per spec? (Otherwise elements with tag name `siblings` would collide with the proposed feature.)

**Comment by mirisuzanne** [comment]
yes, the root selector is in parenthesis. So this pattern would be something like `@scope siblings (.foo) {}` if we go with a keyword approach. (I don't have a better idea for it off-hand)

**Comment by bramus** [comment]
I had this idea of changing the `to` keyword itself to `siblings` and let that be the differentiating factor, e.g. 

- `@scope (.foo) siblings (.bar) { … }`
- `@scope (.foo) siblings {  … }`

Reasons not to like that:
- It’s not clear upfront that you’re going to scope between siblings. You first need to read the upper boundary selector _(which might be long)_ and then come to know you’re doing a siblings-scope.
- Weird to see when no lower boundary is defined

I don’t think an extra at-rule (e.g. `@sibling-scope`) is feasible, to keep the number of at-rules in check?

**Comment by tabatkins** [comment]
I'm actually not sure if there are impl constraints that would encourage us to keep the number of at-rules in check or not. If not, `@sibling-scope` might very well be the most readable variant, clearly suggesting both that it's tied to `@scope`, and that it's different.

**Comment by bramus** [comment]
Agenda+ to decide between `@scope siblings` and `@sibling-scope`. Proposal to go for `@sibling-scope`.

**Comment by andruud** [comment]
> I'm actually not sure if there are impl constraints that would encourage us to keep the number of at-rules in check or not.

I don't think there are such constraints.
